gluten-ut/src/test/scala/org/apache/spark/sql/GlutenDataFramePivotSuite.scala

```diff
@@ -17,9 +17,28 @@
 
 package org.apache.spark.sql
 
+import org.apache.spark.sql.functions._
+
 class GlutenDataFramePivotSuite extends DataFramePivotSuite with GlutenSQLTestsTrait {
 
   override def testNameBlackList: Seq[String] = Seq(
-    GlutenTestConstants.IGNORE_ALL
+    // Replaced with the below test.
+    "pivot with column definition in groupby"
   )
+
+  // This test is ported from vanilla spark with pos value (1-based) changed from 0 to 1 for
+  // substring. In vanilla spark, pos=0 has same effectiveness as pos=1. But in velox, pos=0
+  // will return an empty string as substring result.
```

Comment:
Are we using `substring` under presto sql or spark sql? Maybe we can consider to implement the aligned substring in spark sql.

Reply:
We are using the function in presto sql, where it is following Presto's semantics, e.g, return empty string for pos=0. For vanilla spark, the `pos` is 1-based. It is not a general usage to pass 0 for it. So we currently work around this issue in this test.